### ISSUE TITLE
Install Fails using Apt

### ISSUE THREAD

**Issue by stjagu** [ipartola/hawkeye#4]
Hello there,

When trying to install Hawkeye on RPI  I encounter the following:

```
sudo apt-get install hawkeye
Reading package lists... Done
Building dependency tree
Reading state information... Done
Some packages could not be installed. This may mean that you have
requested an impossible situation or if you are using the unstable
distribution that some required packages have not yet been created
or been moved out of Incoming.
The following information may help to resolve the situation:

The following packages have unmet dependencies:
 hawkeye : Depends: libc6 (>= 2.15) but 2.13-38+rpi2+deb7u8 is to be installed
E: Unable to correct problems, you have held broken packages
```

Any advice on how to resolve?

Thanks

**Comment by ipartola** [comment]
Looks like there's a mismatch with the libc6 library version. What are you running on your RPi? Raspbian? Is it up to date? If you can bring it up to date, that might solve this issue.

If not, you can always rebuild the .deb package for your system. See the README for instructions on how to do this.

**Comment by jepolch** [comment]
I'm running Wheezy from FlightAware (called Piaware) and I'm getting the same error as the original poster.

Linux piaware 3.18.7+ #755 PREEMPT Thu Feb 12 17:14:31 GMT 2015 armv6l GNU/Linux

I would really love to get this working. I'll look at the instructions to see if I can rebuld the deb.

**Comment by jepolch** [comment]
I tried to install Hawkeye on a Pi 2 running Raspbian Wheezy:
Linux raspberrypi 3.18.11-v7+ #776 SMP PREEMPT Mon Apr 6 13:19:43 BST 2015 armv7l GNU/Linux

I received the same error that I did on Piaware Wheezy:
The following packages have unmet dependencies:
 hawkeye : Depends: libc6 (>= 2.15) but 2.13-38+rpi2+deb7u8 is to be installed
E: Unable to correct problems, you have held broken packages.

When I try to install from source it fails with a different error (this happened on two different Pi's):
pi@raspberrypi ~ $ git clone git@github.com:ipartola/hawkeye.git
Cloning into 'hawkeye'...
The authenticity of host 'github.com (192.30.252.129)' can't be established.
RSA key fingerprint is 16:27:ac:a5:76:28:2d:36:63:1b:56:4d:eb:df:a6:48.
Are you sure you want to continue connecting (yes/no)? yes
Warning: Permanently added 'github.com,192.30.252.129' (RSA) to the list of known hosts.
Permission denied (publickey).
fatal: The remote end hung up unexpectedly

**Comment by jepolch** [comment]
I downloaded the source directly from https://github.com/ipartola/hawkeye

I compiled the code and now have Hawkeye running on my Pi 2. 
pi@raspberrypi ~/hawkeye-master/src $ 2015-05-08 23:36:34 INFO: Starting capture on device /dev/video0.
2015-05-08 23:36:34 INFO: Starting server.
2015-05-08 23:36:34 INFO: Listening on 0.0.0.0 port 8000.

'make install' fails, so I have no /etc/hawkeye/hawkeye.conf. I tried viewing the web page with both Chrome and Firefox, but neither was able to display the page. Firefox says: Unable to Connect

**Comment by jepolch** [comment]
Sorry to "bomb" this thread, but I finally got it installed using the basic instructions and I wanted to share how I did it. I had to install unstable packages - not not quick or painless. I found these instructions:

You can use unstable packages. Just add this line to /etc/apt/sources.list:

   deb http://ftp.us.debian.org/debian sid main

To upgrade your packages, run

   apt-get update && apt-get upgrade libc6

Trust me, it makes a LOT of changes to your system, just to upgrade libc6.


Now Hawkeye works, /etc/hawkeye/hawkeye.conf has been installed, but I still can't connect to port 8000 on my local network. I haven't tried tried attaching a monitor to the Pi yet. I'll try that tomorrow.

**Comment by ipartola** [comment]
Glad you got it working though using the unstable release is definitely not the desired way to get this going. I am going to try to replicate your setup and rebuild the binary .deb package so that it's compatible with the modern stable Raspbian.

I also updated the instructions for how to build Hawkeye from source: I was using the wrong URL for the `git clone` command. That was another step in the confusion here.

What exactly did `make install` fail on?

Lastly, try connecting on localhost rather than directly. Make sure your /etc/hawkeye/hawkeye.conf specifies the correct address to listen on. You can get a sample config file in the doc directory or here: https://github.com/ipartola/hawkeye/blob/master/doc/hawkeye.conf

Note that by default it listens on localhost port 8000 which would mean you cannot connect from anything but the RPi. This is a security measure so that you don't open it up to the world with the default settings.

**Comment by jepolch** [comment]
Thanks for taking a look at this. As I said, I would love to get this working. I have motionPie on another Pi, but that's a dedicated install image. You can't install anything else on the Pi that's running motionPie. I want to get Hawkeye going on a Pi that's running a bunch of stuff. Adding a camera to that Pi would be a great bonus.

'make install' said - I think I remember - "nothing to do". I compiled the code just running 'make' and it built the hawkeye binary - which worked - but when I ran 'make install' it wouldn't.

I appreciate your concern about security, but I don't see how anyone could view the server when it's on 192.168.x.x and behind a router. I'll try to view it using tightvnc, but that's not a very convenient option. Thanks again.

By the way, when I do a portscan of the Pi running Hawkeye it doesn't show port 8000.

**Comment by jepolch** [comment]
I installed tightvncserver and when I view the desktop remotely and connect to Hawkeye using the Epiphany browser, it shows nothing. No error messages, just a blank white screen. This is a USB 2.0 camera I've used with motionPie in the past. Here's the output from 'lsusb'.

Bus 001 Device 004: ID 0ac8:3420 Z-Star Microelectronics Corp. Venus USB2.0 Camera

http://i62.tinypic.com/2pzjl9u.png

**Comment by AbeEstrada** [comment]
A better way to just upgrade is: `# apt-get install --only-upgrade libc6`

**Comment by ipartola** [comment]
I spent some time over the weekend working on this but it's still not clear what's wrong. The specific issue with libc6 version is pretty easy: I needed to rebuild the package on a Raspbian RPi box, not an RPi running a different distro. That's my mistake. However, having done that, I am running into multiple issues with streaming the video. I will investigate what's up and update this ticket when I have a new working build.
